src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -0,0 +1,151 @@
+package org.mifos.connector.mpesa.camel.routes;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.json.JSONObject;
+import org.mifos.connector.mpesa.auth.AccessTokenStore;
+import org.mifos.connector.mpesa.dto.BuyGoodsPaymentRequestDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.util.Base64;
+
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.ACCESS_TOKEN;
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.BUY_GOODS_REQUEST_BODY;
+import static org.mifos.connector.mpesa.safaricom.config.SafaricomProperties.MPESA_BUY_GOODS_TRANSACTION_TYPE;
+
+
+@Component
+public class SafaricomRoutesBuilder extends RouteBuilder {
+
+    @Value("${mpesa.api.passKey}")
+    private String passKey;
+
+    @Value("${mpesa.api.lipana}")
+    private String buyGoodsBaseUrl;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AccessTokenStore accessTokenStore;
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public void configure() {
+
+        /**
+         * Rest endpoint to initiate payment for buy goods
+         *
+         * Sample request body: {
```

Comment:
`Timestamp`, `Password` and `TransactionType` are generated during runtime and need not to be passed by user as a body parameter. Because

`TransactionType`: is always same i.e **CustomerBuyGoodsOnline**
`Password`: base64(businessShortCode + passKey + timestamp)